cloudrunservice.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"log"
+)
+
+// Maximum length of a Cloud Run Service name.
+const maxCloudRunServiceNameLen = 53
+
+// CloudRunService represents a Cloud Run Service and holds its name and URL.
+type CloudRunService struct {
+	// the associated Sample
+	sample *Sample
+
+	// the Service's name
+	name string
+
+	// the Service's root URL
+	url string
+}
+
+// newCloudRunService returns a pointer to a new CloudRunService with the provided name.
```

Comment:
nit: No need to clarify it's a pointer 
```suggestion
// newCloudRunService returns a new CloudRunService with the provided name.
```